### ISSUE TITLE
tlsNormalize failing to normalize TLS data

### ISSUE THREAD

**Issue by munachau** [tiagodc/TreeLS#55]
I used the TLS data available in package pine_plot.laz. but it is unable to normalize as it is related with function raster

tls = tlsNormalize(tls, keep_ground = T)
no ground points found, performing ground segmentation
Error in h(simpleError(msg, call)) : 
  error in evaluating the argument 'x' in selecting a method for function 'raster': unable to find an inherited method for function ‘extent’ for signature ‘"LAS"’

**Comment by ashlynolah** [comment]
I am also having this issue.

**Comment by green512** [comment]
Me too

**Comment by bi0m3trics** [comment]
It's line 545 of [methods.R](https://github.com/tiagodc/TreeLS/blob/0e0871f11358b2812adc7564ec66c39a5ab70c71/R/methods.R)  - `grid = las %>% extent %>% raster` which is trying to apply `extent` to an object of class `LAS` and instead `lidR` uses `ext` from `terra` since `lidR` no longer uses `raster `(as no one should, since the "[package has been superseded by the "terra" package](https://cran.r-project.org/web/packages/raster/index.html)")

In short, I'd be careful even using this package at all as it's no longer supported, but if you do I suggest rewriting functions (like `tlsNormalize`) to use them locally (or fork the repo and make your on version of the package) where objects like rasters produced from function in `lidR `are converted to `raste::raster` prior to being used. That or undertake the epic task of reworking the entire package to use `terra `instead of `raster`.

**Comment by Estrada-Gene** [comment]
thank you @bi0m3trics for the helpful response. I tried using your updated tlsNormalize() function as you've written it out above, but I get this error: could not find function "isLAS". Is this function in a different package that I should load or is this just example code that I should adjust myself for my own needs?

**Comment by bi0m3trics** [comment]
@Estrada-Gene isLAS is part of TreeLS, so if you know it's a las you can delete it...

**Comment by spono** [comment]
@bi0m3trics I forked the project and moved `sp/raster` functions to `sf/terra`...it still may need some proof-check but it runs the examples pretty smoothly :)
If any of you can test it a bit against the previous version, I can then push a pull request to the main repo and let TreeLS roll again

**Comment by radt0005** [comment]
Thanks @spono and @bi0m3trics.

**Comment by crespelp** [comment]
Same problem, and I´m too neophyte to try the suggested fixes. Is this library current?

**Comment by bi0m3trics** [comment]
@crespelp It was current to R 4.3 about a year ago. @spono has a fork that was current a few months ago (correct me if I'm wrong, Nic) and my fork was current to somewhere in between (but definitely pre-4.4)... so this repo - maybe no, other forks - maybe yes. Not a great answer, but the truth.

I suggest installing Nic's fork, and seeing if it works...
remotes::install_github('spono/TreeLS')

**Comment by spono** [comment]
@profstephan I quickly checked my fork and it seems to be still running well (at least the demo lines from the front page of the package).

Try to install it using bi0m3trics's suggestion:
`remotes::install_github('spono/TreeLS')`

BTW: this package is an easy tool for teaching, if that is your target...but I would suggest to keep an eye on FORTLS and lidR (recently developing also tools for TLS analysis)
